src/datomish_user_agent_service/server.cljs

```diff
@@ -1,229 +1,405 @@
-                   (.json res (clj->js {})))))))
+                 (if-not (.is req "application/edn") ;; TODO: accept application/json?
+                   (doto res
+                     (.status 415)
+                     (.json (clj->js {:error "Unsupported Media Type"})))
+                   (let [body    (aget req "body")
+                         edn     (cljs.reader/read-string body)
+                         tx-data (:tx-data edn)
+                         report  (<? (d/<transact! (<? connection-pair-chan) tx-data))
+                         results (select-keys report [:tx-data :tempids])]
+                     (cond
+                       (.accepts req "json")
+                       (doto res
+                         (.json (clj->js results)))
 
-    (.get "/query"
-          (auto-caught-route-error
-            (fn [req]
-              (-> req
-                  (.checkQuery "q")
-                  (.notEmpty))
-              (-> req
-                  (.checkQuery "limit")
-                  (.optional)
-                  (.isInt))
-              (-> req
-                  (.checkQuery "since")
-                  (.optional)
-                  (.isInt))
-              (-> req
-                  (.checkQuery "snippetSize")
-                  (.optional))
-              )
-            (fn [req res]
-              (go-pair
-                (let [results (<? (api/<saved-pages-matching-string (d/db (<? connection-pair-chan)) ;; TODO -- unify on conn over db?
-                                                                    (aget req "query" "q")
-                                                                    ;; {:limit (int (or (-> req .-query .-limit) 100))} ;; TODO - js/Number.MAX_SAFE_INTEGER
-                                                                    ))]
-                  (.json res (clj->js {:results results})))))))))
+                       (.accepts req "application/edn")
+                       (doto res
+                         (.set "Content-Type" "application/edn") ;; TODO: charset?
+                         (.send (prn-str results)))
+
+                       :default
+                       (doto res
+                         (.status 406)
+                         (.send #js {:error "Not Acceptable"}))
+                       )))))))
+    ))
+
+(defn- v1-router [connection-pair-chan]
+  (let [ws-clients
+        (atom #{})
+
+        diff
+        (fn [type payload]
+          (println "Sending diff message of type" type "and payload" payload "to" (count @ws-clients) "clients")
+          (let [message (js/JSON.stringify (clj->js {:message "diff" :type type :payload payload}))]
+            (go-pair
+              (doseq [ws @ws-clients]
+                (.send ws message)))))
+
+        send-bookmark-diffs
+        (fn []
+          (go-promise
+            identity
+
+            (let [results (<? (api/<starred-pages (d/db (<? connection-pair-chan)) ;; TODO -- unify on conn over db?
```

Comment:
Still a worthwhile abstraction to keep — `db` emphasizes that you're querying a consistent snapshot.